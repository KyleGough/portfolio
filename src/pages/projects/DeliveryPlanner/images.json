[
  { "src": "/img/delivery-route-planner1.jpg", "caption": "Delivery view" },
  { "src": "/img/delivery-route-planner2.jpg", "caption": "Route view" },
  { "src": "/img/delivery-route-planner3.jpg", "caption": "Generated report" },
  { "src": "/img/delivery-route-planner4.jpg", "caption": "Generated route" }
]
