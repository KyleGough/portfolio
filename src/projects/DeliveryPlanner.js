import React, { useEffect } from 'react';
import Section from '../components/Section';
import Divider from '../components/Divider';
import Chip from '../components/Chip';
import Pagination from '../components/Pagination';
import Figure from '../components/Figure';
import Screenshots from '../components/Screenshots';

export default function DeliveryPlanner() {

    useEffect(() => {
        document.title = 'Delivery Route Planner';
    }, []);

    const images = [
        { src:'/img/delivery-route-planner1.jpg', caption:'Delivery view' },
        { src:'/img/delivery-route-planner2.jpg', caption:'Route view' },
        { src:'/img/delivery-route-planner3.jpg', caption:'Generated report' },
        { src:'/img/delivery-route-planner4.jpg', caption:'Generated route' }
    ];

    return (
        <>
        <Section>
            <h1 className='project-title'>Delivery Route Planner</h1>
            <h2 className='project-subtitle'>A-Level Computing Coursework</h2>
            <p className='text-link-hover my-4'>December 2015 - March 2016</p>
            <p className='mb-4'>Software written in C# which creates an efficient route between multiple delivery locations. Uses SQL to store and query data on products, clients and their deliveries. Produces a report which communicates the route, all items in the deliveries, and the estimated time taken. Uses graph algorithms such as Dijkstra's and Nearest Neighbour with 2-Opt to optimise the route.</p>

            <div className='flex flex-row flex-wrap items-center mt-8'>
                <Chip name='C#' />
                <Chip name='SQL' />
            </div>
        </Section>

        <Divider />

        <Figure image={images[0]} />

        <Divider />

        <Section>
            <h2 className='project-header'>Features</h2>
            <ul className='project-list'>
                <li>Creates an efficient route for a number of selected deliveries.</li>
                <li>Utilises database to store client, delivery, and product data.</li>
                <li>Ability to view/add/edit/delete client, delivery and product data.</li>
                <li>Creates a printable report including delivery order, total items and estimated time.</li>
            </ul>
        </Section>

        <Divider />

        <Screenshots images={images} />

        <Divider />
        
        <Pagination
            previousTitle='Sorting Algorithm Visualiser'
            previousLink='/projects/sorting-algorithm-visualiser'
            nextTitle='To Do List'
            nextLink='/projects/todo-list'
        />
        </>
    );
}