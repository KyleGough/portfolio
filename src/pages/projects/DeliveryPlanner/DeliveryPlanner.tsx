import React, { useEffect } from 'react';

import { Divider } from '../../../components/Divider';
import { ImageFigure } from '../../../components/ImageFigure';
import { Pagination } from '../../../components/Pagination';
import { ProjectHeader } from '../../../components/ProjectHeader';
import { Screenshots } from '../../../components/Screenshots';
import { Section } from '../../../components/Section';
import { getProjectData } from '../../../data';
import images from './images.json';

const project = getProjectData('delivery-route-planner');

export const DeliveryPlanner: React.FC = () => {
  useEffect(() => {
    document.title =
      'Delivery Route Planner - Delivery Route and Report Generation Software';
  }, []);

  return (
    <>
      <ProjectHeader project={project} />

      <Divider />

      <ImageFigure image={images[0]} />

      <Divider />

      <Section>
        <h2 className="project-header">Features</h2>
        <ul className="project-list">
          <li>
            Creates an efficient route for a number of selected deliveries.
          </li>
          <li>
            Utilises database to store client, delivery, and product data.
          </li>
          <li>
            Ability to view/add/edit/delete client, delivery and product data.
          </li>
          <li>
            Creates a printable report including delivery order, total items and
            estimated time.
          </li>
        </ul>
      </Section>

      <Divider />

      <Screenshots images={images} />

      <Divider />

      <Pagination
        previousTitle="Sorting Algorithm Visualiser"
        previousLink="/projects/sorting-algorithm-visualiser"
        nextTitle="To Do List"
        nextLink="/projects/todo-list"
      />
    </>
  );
};
