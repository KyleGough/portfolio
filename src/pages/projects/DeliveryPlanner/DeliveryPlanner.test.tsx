import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { DeliveryPlanner } from './DeliveryPlanner';

describe('DeliveryPlanner project page', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <DeliveryPlanner />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
