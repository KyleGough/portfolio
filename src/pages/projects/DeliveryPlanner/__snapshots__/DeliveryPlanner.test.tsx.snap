// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryPlanner project page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      Delivery Route Planner
    </h1>
    <h2
      class="project-subtitle"
    >
      A-Level Computing Coursework
    </h2>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2015-12"
      >
        December 2015
      </time>
       
      -
       
      <time
        datetime="2016-03"
      >
        March 2016
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      Creates an efficient route between multiple delivery locations. Store, view and query data on products, clients and their deliveries. Produces a report which communicates the route, delivery items, and ETA. Uses graph algorithms such as Dijkstra's and Nearest Neighbour with 2-Opt to optimise the route.
    </p>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        C#
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        SQL
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Delivery view"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/delivery-route-planner1.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Delivery view
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Features
    </h2>
    <ul
      class="project-list"
    >
      <li>
        Creates an efficient route for a number of selected deliveries.
      </li>
      <li>
        Utilises database to store client, delivery, and product data.
      </li>
      <li>
        Ability to view/add/edit/delete client, delivery and product data.
      </li>
      <li>
        Creates a printable report including delivery order, total items and estimated time.
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Delivery view"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/delivery-route-planner1.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Delivery view
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Route view"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/delivery-route-planner2.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Route view
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Generated report"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/delivery-route-planner3.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Generated report
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Generated route"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/delivery-route-planner4.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Generated route
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/sorting-algorithm-visualiser"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            Sorting Algorithm Visualiser
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/todo-list"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            To Do List
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
